.github/workflows/publish-rule-book.yaml

```diff
@@ -0,0 +1,114 @@
+name: Publish Rule Book
+
+on:
+  # push:
+  #   branches:
+  #     - main
+  pull_request:
+    branches: [main]
+
+env:
+  BOARD_GAME_GEEK_RULE_BOOK: https://boardgamegeek.com/filepage/272787
+  PO4A_VERSION: "0.71"
+  PO4A_GH_URL: https://github.com/mquinson/po4a/releases/download
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [en, pl]
+        include:
+          - language: en
+            version: Version
+            # warning: ""
+          - language: pl
+            version: Wersja
+            # warning: "To jest nieopublikowana wersja zbudowana na GitHubie z commita ${GITHUB_SHA::7}"
+
+    steps:
+      - name: Set up repository
+        uses: actions/checkout@v4
+
+      - name: Replace rule book version with a warning
+        run: |
+          VERSION=$(grep -P "^[\ ]*${{ matrix.version }} \d{1,}.\d{1,}$" main_${{ matrix.language }}.tex)
+          RELEASE_PAGE="${{ github.server_url }}/${{ github.repository }}/releases"
+          WARNING="This is an unreleased version built on GitHub from commit ${GITHUB_SHA::7}. To download the latest official release, please go to \\\href{${RELEASE_PAGE}}{GitHub release page} or \\\href{${{ env.BOARD_GAME_GEEK_RULE_BOOK }}}{BoardGameGeek files section}."
+          sed -i "s@$VERSION@$WARNING@g" main_${{ matrix.language }}.tex
+
+      - name: Generate translations for ${{ matrix.language }}
+        if: ${{ matrix.language != 'en' }}
+        run: |
+          sudo apt-get install -y po4a  # required as standalone po4a is missing some deps
+          wget --quiet ${{ env.PO4A_GH_URL }}/v${{ env.PO4A_VERSION }}/po4a-${{ env.PO4A_VERSION}}.tar.gz  # sadly the version from apt isn't working
+          export PO4A_DIR="po4a-${{ env.PO4A_VERSION }}"
+          tar -xf ${PO4A_DIR}.tar.gz
+          export PERL5LIB="${PO4A_DIR}/lib:${PERL5LIB}"
+          ./${PO4A_DIR}/po4a -v --no-update po4a.cfg
```

Comment:
I see lots of code is duplicated between build* and publish* workflows.

Maybe instead add some "if" conditions on the publishing steps here, and remove the other one?

Reply:
True, I can use different triggers for the workflow. Let me check that.

Reply:
No repetition any more.